src/soorgeon/cli.py

```diff
@@ -112,41 +117,69 @@ def test(filename):
         nb = jupytext.read(filename)
         # convert ipynb to py and create a temp file in current directory
-        directory = dirname(abspath(filename))
-        with tempfile.NamedTemporaryFile(suffix=".py",
-                                                delete=True,
-                                                dir=directory) as temp_file:
+        with tempfile.NamedTemporaryFile(suffix=".ipynb",
+                                         delete=True,
+                                         dir=directory) as temp_file:
             jupytext.write(nb, temp_file.name)
-            _test(temp_file.name)
+            _test(temp_file.name, output_filename)
     else:
-        _test(filename)
+        _test(filename, output_filename)
 
 
-def _test(filename):
+def _test(filename, output_filename):
     try:
-        exec(open(filename).read())
-        click.echo(f"Finished executing {filename}, no error encountered")
-    except (ModuleNotFoundError, AttributeError, SyntaxError) as err:
+        pm.execute_notebook(filename, output_filename, kernel_name='python3')
+        click.secho(f"""\
+Finished executing {filename}, no error encountered.
+Output notebook: {output_filename}""",
+                    fg='green')
+    except PapermillExecutionError as err:
+        error_traceback = err.traceback
         error_suggestion_dict = {
-            "ModuleNotFoundError": "create a virtualenv, and"
-            " adding a requirements.txt with the package",
-            "AttributeError": "downgrade some libraries",
-            "SyntaxError": "check syntax",
+            "ModuleNotFoundError":
+            "Some packages are missing, please install them "
+            "with 'pip install {package-name}'\n"
+            "\nContact us for help: https://ploomber.io/community",
+            "AttributeError":
+            "AttributeErros might be due to changes in the libraries "
+            "you're using. "
+            "\nContact us for help: https://ploomber.io/community",
+            "SyntaxError":
+            "There are syntax errors in the notebook. "
+            "\nContact us for help: https://ploomber.io/community",
         }
-        error_type = type(err).__name__
-        click.echo(f"""
-        {error_type} encountered while executing the notebook: {err}
-
-        It is recommended to {error_suggestion_dict[error_type]}
-        """)
+        for error, suggestion in error_suggestion_dict.items():
+            if any(error in error_line for error_line in error_traceback):
+                click.secho(f"""\
+{error} encountered while executing the notebook: {err}
+{suggestion}
+Output notebook: {output_filename}""",
+                            fg='red')
+                return
+
+        click.secho(f"""\
+Error encountered while executing the notebook: {err}
+
+Contact us for help: https://ploomber.io/community
+
+Output notebook: {output_filename}""",
+                    fg='red')
     except Exception as err:
```

Comment:
in these `except` statements you're not raising an exception, this means that even if there is an error, the exit code will be 0, this will make CI systems interpret this as a successful execution, to fix this:

```python
err = None

# the try except code, on each except statement do
except Exception as e:
  err = e


at the end:

if err is not None:
   # raise the exception here
```

Since you're customizing the message depending on the scenario, you can split the error message into two:

```python
from click.exceptions import ClickException

click.secho('custom message', fg='red')
raise ClickException('An error happened when executing the notebook, contact us...')
```